apps/src/netsim/NetSimSendWidget.js

```diff
@@ -48,30 +81,137 @@ NetSimSendWidget.createWithin = function (element, connection) {
 };
 
 /**
+ * Creates a keyPress handler that allows only the given characters to be
+ * typed into a text field.
+ * @param {RegExp} whitelistRegex
+ * @return {function} appropriate to pass to .keypress()
+ */
+var whitelistCharacters = function (whitelistRegex) {
```

Comment:
[Here's the style guideline](http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml?showone=Function_Declarations_Within_Blocks#Function_Declarations_Within_Blocks).  There doesn't seem to be a rule for top-level, but the rule for declarations within blocks is that you must *not* use `function name() {`.

Now, because of our build pipeline, even declarations that are top-level within a file are technically scoped inside of a module.  We can choose to rule on this styling at a file-scope, but I don't think that's ECMA-standard unless our build converts the declaration.